src/engine/sys/sdl_glimp.cpp

```diff
@@ -2285,77 +2286,84 @@ static void GLimp_InitExtensions()
-				{ "2.0", { "2.0-devel", "2.0-rc1", } },
-			};
+				std::string mesaVersion = "";
+
+				static const std::pair<std::string, std::vector<std::string>> versions[] = {
+					{ "1.0", { "1.0-devel", "1.0-rc1", "1.0-rc2", "1.0-rc3", "1.0-rc4", "1.0", } },
+					{ "1.",  { "1.1", "1.2", "1.3", "1.4", } },
+					{ "2.0", { "2.0-devel", "2.0-rc1", } },
+				};
 
-			for ( auto& vp : versions ) {
-				mesaVersion = Str::Format( "Mesa 24.%s", vp.first );
+				for ( auto& vp : versions ) {
+					mesaVersion = Str::Format( "Mesa 24.%s", vp.first );
 
-				if ( Q_stristr( glConfig.version_string, mesaVersion.c_str() ) ) {
-					for ( auto& v : vp.second ) {
-						mesaVersion = Str::Format( "Mesa 24.%s", v );
+					if ( Q_stristr( glConfig.version_string, mesaVersion.c_str() ) ) {
+						for ( auto& v : vp.second ) {
+							mesaVersion = Str::Format( "Mesa 24.%s", v );
 
-						if ( Q_stristr( glConfig.version_string, mesaVersion.c_str() ) ) {
-							foundMesa241 = true;
-							break;
+							if ( Q_stristr( glConfig.version_string, mesaVersion.c_str() ) ) {
+								foundMesa241 = true;
+								break;
+							}
 						}
-					}
 
-					break;
+						break;
+					}
 				}
-			}
 
-			if ( foundMesa241 ) {
-				logger.Warn( "...found buggy %s driver, ARB_bindless_texture disabled", mesaVersion );
+				if ( foundMesa241 ) {
+					logger.Warn( "Found buggy %s driver, disabling ARB_bindless_texture.", mesaVersion );
+					bindlessTextureEnabled = false;
+				}
 			}
-		}
-
-		bool foundOglp = false;
 
-		if ( workaround_glDriver_amd_oglp_disableBindlessTexture.Get() )
-		{
 			/* AMD OGLP driver for Linux shares the same vendor string than AMD Adrenalin driver
 			for Windows and AMD ATI driver for macOS. When running the Windows engine binary on
 			Wine we must check we're not running Windows or macOS to detect Linux OGLP. */
-			if ( Q_stristr( glConfig.vendor_string, "ATI Technologies Inc." ) )
+			if ( workaround_glDriver_amd_oglp_disableBindlessTexture.Get() )
 			{
-				#if defined(_WIN32)
-					// Detect Wine being used.
-					if ( Sys::isRunningOnWine() )
-					{
-						// Detect AMD ATI driver on macOS not being used.
-						if ( !Q_stristr( glConfig.version_string, "ATI-" ) )
+				bool foundOglp = false;
+
+				if ( glConfig2.hardwareVendor == glHardwareVendor_t::ATI
+					&& glConfig2.driverVendor != glDriverVendor_t::MESA )
+				{
+					#if defined(_WIN32)
+						// Detect Wine being used.
+						if ( Sys::isRunningOnWine() )
 						{
-							foundOglp = true;
+							// Detect AMD ATI driver on macOS not being used.
+							if ( !Q_stristr( glConfig.version_string, "ATI-" ) )
+							{
+								foundOglp = true;
+							}
 						}
-					}
-				#elif defined(__APPLE__)
-					// AMD ATI driver for macOS is believed to not implement bindless texture.
-				#else
-					foundOglp = true;
-				#endif
-			}
+					#elif defined(__APPLE__)
+						// AMD ATI driver for macOS is believed to not implement bindless texture.
```

Comment:
AFAIK Apple is stuck on 4.1 completely due to their petty bullshit of trying to push Metal.

Reply:
Yes bindless texture does not work on macOS and probably never will.

Reply:
Maybe with Mesa zink over MoltenVK over Metal, but then it's Mesa, and then not the Apple AMD/ATI driver. 😄️

Reply:
> Yes

Thanks for the confirmation.

This can then set `foundOglp` to `true`.

Reply:
From dev channel:
> in fact, iirc, there is even a warning printed in the console about missing bindless texture extension when you launch unv on a mac

So looks like we don't even need to special-case Mesa on Apple here.

Reply:
What I did is that I now detect what is the host system running wine. I don't test for anything being Apple anymore.